protocol-units/mempool/move-rocks/src/lib.rs

```diff
@@ -332,18 +363,32 @@ impl MempoolTransactionOperations for RocksdbMempool {
 			let cf_handle = db
 				.cf_handle(cf::MEMPOOL_TRANSACTIONS)
 				.ok_or_else(|| Error::msg("CF handle not found"))?;
+			let timeline_cf_handle = db
+				.cf_handle(cf::TRANSACTION_TIMELINE)
+				.ok_or_else(|| Error::msg("CF handle not found"))?;
 			let lookups_cf_handle = db
 				.cf_handle(cf::TRANSACTION_LOOKUPS)
 				.ok_or_else(|| Error::msg("CF handle not found"))?;
 			let mut read_options = ReadOptions::default();
 			read_options
-				.set_iterate_upper_bound(construct_timestamp_threshold_key(timestamp_threshold)?);
-			let mut iter = db.iterator_cf_opt(&cf_handle, read_options, IteratorMode::Start);
+				.set_iterate_upper_bound(construct_timeline_threshold_key(timestamp_threshold));
+			let mut iter =
+				db.iterator_cf_opt(&timeline_cf_handle, read_options, IteratorMode::Start);
 			let mut transaction_count = 0;
 			let mut batch = WriteBatch::default();
 
-			if let Some(res) = iter.next() {
-				let (key, value) = res?;
+			while let Some(res) = iter.next() {
+				let (timeline_key, key) = res?;
+
+				batch.delete_cf(&timeline_cf_handle, &timeline_key);
```

Comment:
Two things w.r.t. to this implementation and testing that would be nice to haves:
1. I think we might want to fuzz a fair few of these tests--particularly the GC. 
2. I think we would benefit from using domain-specific types for GC timestamps. I was looking at the usage here and in the other garbage collected containers we have, and it's quite easy to mix up the timestamps with something else.

Reply:
Sounds reasonable. As we need custom formatting here, perhaps a newtype defined in the library and implementing `Display` to format into `{:032}` would be appropriate.